### PR TITLE
Migrate goal account participation commands to REST API

### PR DESCRIPTION
**Pull request by winder**
## Summary

Use REST API instead of reading partkey files from the data directory in goal commands.

The changes here were very small right up until I modified `chooseParticipation`, which had a bunch of upstream and downstream effects.

## Test Plan

New e2e test focusing on the goal commands.

Existing e2e tests leverage the libgoal changes pretty heavily.

### CODE REVIEW

**Comment by tsachiherman** [comment]
could you please rebase that on master ?

**Comment by winder** [comment]
@tsachiherman ok, it should be ready now.

File: test/e2e-go/features/transactions/onlineStatusChange_test.go
Comment (tsachiherman) [inline test/e2e-go/features/transactions/onlineStatusChange_test.go:172]: I think that we'll need a bit more than that : 
```
The directory is neither guaranteed to exist nor have accessible permissions.
```

Comment (winder) [reply test/e2e-go/features/transactions/onlineStatusChange_test.go]: Changed to:
```
	tempDir, err := ioutil.TempDir(os.TempDir(), "test-close-on-error")
	require.NoError(t, err)
	defer os.RemoveAll(tempDir)
```

**Review by tsachiherman — APPROVED** [review]
looks good to me.

**Comment by winder** [comment]
One of the new tests had a failure that I need to investigate and see why it seems to be flaky:
```
whole log follows (/opt/cibuild/project/test/scripts/e2e_subs/goal-partkey-commands.sh):
+ date '+/opt/cibuild/project/test/scripts/e2e_subs/goal-partkey-commands.sh start %Y%m%d_%H%M%S'
/opt/cibuild/project/test/scripts/e2e_subs/goal-partkey-commands.sh start 20220107_191841
+ gcmd='goal -w 56235432CB9E3D872F06633EE1F3D1E4'
++ awk '{ print $3 }'
++ goal -w 56235432CB9E3D872F06633EE1F3D1E4 account list
+ INITIAL_ACCOUNT=5YXRTMVWL5XQ4Q4SBOJ4DTXCOIK52NZQFSUQNMA6NFVJVBFVKLPADWB64I
++ tr -s ' '
++ tail -n 1
++ goal account listpartkeys
+ OUTPUT='no HB3F...ETHU CN3FWZ6J... N/A 0 120'
+ [[ no HB3F...ETHU CN3FWZ6J... N/A 0 120 != yes* ]]
+ fail_test 'Registered should be '\''yes'\'' but wasn'\''t.'
+ echo -e 'Registered should be '\''yes'\'' but wasn'\''t.'
Registered should be 'yes' but wasn't.
+ exit 1
```

**Comment by winder** [comment]
I thought I left a comment here, but it seems I never submitted it.

The flaky test seems to occur on master as well, so there are no technical blockers to prevent this PR from being merged.

However, I understand that there are some merge conflicts with the state proofs project. Given the current schedule, @algoidan and @algonoah have requested holding off on merging this PR until after that project is merged in.

**Pull request closed without merging**
